bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/SamsungTvTlsTrustManagerProvider.java

```diff
@@ -18,35 +18,52 @@
 
 import javax.net.ssl.SSLEngine;
 import javax.net.ssl.X509ExtendedTrustManager;
+import javax.security.auth.x500.X500Principal;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.smarthome.io.net.http.TlsTrustManagerProvider;
 import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Provides a TrustManager to allow secure websocket connections to any TV (=server)
+ * TODO: convert to using TrustAllTrustMananger (not supported in OH2.4 yet)
  *
  * @author Arjan Mels - Initial Contribution
  */
 @Component
 @NonNullByDefault
 public class SamsungTvTlsTrustManagerProvider implements TlsTrustManagerProvider {

```

Comment:
Sorry, maybe it was not clear from my comment: loggers should be `private final`, in some rare cases (logging from static methods) `private static final`. You did remove the final again, it allows for better optimization because the compiler knows this will never change.

Reply:
Ah ok, I checked another class and there it was also non final. I will update all. However I found another small bug while testing, so please hold this pull request for the moment.

Reply:
Ok I fixed the finals and one bug, however I still have a problem with repeated discovery that I am looking into.

Reply:
Ok, finally got round to more testing and fixing.
Should be good to go from my perspective now.
@J-N-K can you remove the "work in progress" label?